<launch>
  <arg name="gui" default="true"/>

  <node name="image_publisher"
        pkg="jsk_perception" type="image_publisher.py">
    <rosparam subst_value="true">
      file_name: $(find jsk_perception)/sample/kiva_pod_mask.jpg
      encoding: mono8
      publish_info: false
    </rosparam>
  </node>

  <node name="snake_segmentation"
        pkg="nodelet" type="nodelet"
        args="standalone jsk_perception/SnakeSegmentation">
    <remap from="~input" to="image_publisher/output"/>
  </node>

  <group if="$(arg gui)">
    <node name="tile_image"
          pkg="jsk_perception" type="tile_image.py">
      <rosparam>
        input_topics:
          - /image_publisher/output
          - /snake_segmentation/debug
        draw_topic_name: true
        font_scale: 3.0
      </rosparam>
    </node>

    <node name="image_view"
          pkg="image_view" type="image_view">
      <remap from="image" to="tile_image/output"/>
    </node>
  </group>
</launch>
